internal/app/types/friend.go

```diff
@@ -1,7 +1,16 @@
 package types
 
+import (
+	"github.com/globalsign/mgo/bson"
+)
+
 // Friend hold information of a friend
 type Friend struct {
-	ID   string `json:"id" bson:"id"`
-	Name string `json:"name"`
+	Id    bson.ObjectId `bson:"_id" json:"id"`
```

Comment:
consider not to do this as your implementation assume that we will use mongodb for entire our life :)

Reply:
Fixed